radiation/python/radlw/lwrad_gt4py.py

```diff
@@ -0,0 +1,1800 @@
+    "cldliq",
+    "refliq",
+    "cldice",
+    "refice",
+    "index",
+    "ia",
+    "lcloudy",
+    "tem1",
+    "summol",
+    "lcf1",
+    "cldsum",
+    "tlvlfr",
+    "tlyrfr",
+    "plog",
+    "fp",
+    "ft",
+    "ft1",
+    "indlay",
+    "indlev",
+    "jp1",
+    "tzint",
+    "stempint",
+    "tavelint",
+    "laytrop",
+]
+
+locvars_int = [
+    "ib",
+    "ind0",
+    "ind0p",
+    "ind1",
+    "ind1p",
+    "inds",
+    "indsp",
+    "indf",
+    "indfp",
+    "indm",
+    "indmp",
+    "js",
+    "js1",
+    "jmn2o",
+    "jmn2op",
+    "jpl",
+    "jplp",
+    "id000",
+    "id010",
+    "id100",
+    "id110",
+    "id200",
+    "id210",
+    "id001",
+    "id011",
+    "id101",
+    "id111",
+    "id201",
+    "id211",
+    "jmo3",
+    "jmo3p",
+    "jmco2",
+    "jmco2p",
+    "jmco",
+    "jmcop",
+    "jmn2",
+    "jmn2p",
+    "NGB",
+]
+locvars_flt = [
+    "taug",
+    "pp",
+    "corradj",
+    "scalen2",
+    "tauself",
+    "taufor",
+    "taun2",
+    "fpl",
+    "speccomb",
+    "speccomb1",
+    "fac001",
+    "fac101",
+    "fac201",
+    "fac011",
+    "fac111",
+    "fac211",
+    "fac000",
+    "fac100",
+    "fac200",
+    "fac010",
+    "fac110",
+    "fac210",
+    "specparm",
+    "specparm1",
+    "specparm_planck",
+    "ratn2o",
+    "ratco2",
+]
+
+locvars_rtrnmc = [
+    "clrurad",
+    "clrdrad",
+    "toturad",
+    "totdrad",
+    "gassrcu",
+    "totsrcu",
+    "trngas",
+    "efclrfr",
+    "rfdelp",
+    "fnet",
+    "fnetc",
+    "totsrcd",
+    "gassrcd",
+    "tblind",
+    "odepth",
+    "odtot",
+    "odcld",
+    "atrtot",
+    "atrgas",
+    "reflct",
+    "totfac",
+    "gasfac",
+    "flxfac",
+    "plfrac",
+    "blay",
+    "bbdgas",
+    "bbdtot",
+    "bbugas",
+    "bbutot",
+    "dplnku",
+    "dplnkd",
+    "radtotu",
+    "radclru",
+    "radtotd",
+    "radclrd",
+    "rad0",
+    "clfm",
+    "trng",
+    "gasu",
+    "itgas",
+    "ittot",
+    "ib",
+]
+
+indict = dict()
+for var in invars:
+    if var[-3:] == "tbl":
+        tmp = serializer2.read(var, serializer2.savepoint["lwrad-rtrnmc-input-000000"])
+    else:
+        tmp = serializer.read(var, serializer.savepoint["lwrad-in-000000"])
+
+    if var in ["semis", "icsdlw", "tsfg", "de_lgth"]:
+        indict[var] = np.tile(tmp[:, None, None], (1, 1, nlp1))
+    elif var == "faerlw":
+        tmp2 = np.insert(tmp, 0, 0, axis=1)
+        indict[var] = np.tile(tmp2[:, None, :, :, :], (1, 1, 1, 1, 1))
+    elif var == "gasvmr" or var == "clouds":
+        tmp2 = np.insert(tmp, 0, 0, axis=1)
+        indict[var] = np.tile(tmp2[:, None, :, :], (1, 1, 1, 1))
+    elif var in ["plyr", "tlyr", "qlyr", "olyr", "dz", "delp"]:
+        tmp2 = np.insert(tmp, 0, 0, axis=1)
+        indict[var] = np.tile(tmp2[:, None, :], (1, 1, 1))
+    elif var in ["plvl", "tlvl"]:
+        indict[var] = np.tile(tmp[:, None, :], (1, 1, 1))
+    elif var[-3:] == "tbl":
+        indict[var] = np.tile(tmp[None, None, None, :], (npts, 1, nlp1, 1))
```

Comment:
for passing this over to us when your internship wraps up, it could be helpful to have comments explaining what the processing here is doing (e.g. what shape the data is coming in and what it is being converted to). We can figure it out of course, but comments would be so nice.

Reply:
Will do